driver.go

```diff
@@ -241,14 +241,29 @@ func (d *Driver) Create() error {
 			return errors.Wrap(err, "could not read ssh public key")
 		}
 
-		keyopts := hcloud.SSHKeyCreateOpts{
-			Name:      d.GetMachineName(),
-			PublicKey: string(buf),
+		// Try to find a key with the same Fingerprint
```

Comment:
In general: given the amount of logic here, I'd almost opt to put this in a separate function